src/index.ts

```diff
@@ -1,3 +1,23 @@
-/**
- * DO THE WORK HERE
- */
\ No newline at end of file
+import { fetch } from "undici";
+import { CONSTANTS, ScorecardResult } from "./utils.js";
+
+export async function result(repository: string): Promise<ScorecardResult> {
+  const { href } = buildURL(repository);
+  const response = await fetch(href);
+  const data = (await response.json()) as ScorecardResult;
+
+  return data;
+}
+
+export async function badge(repository: string): Promise<ScorecardResult> {
+  const { href } = buildURL(`${repository}/badge`);
+  const response = await fetch(href);
+  const data = (await response.json()) as ScorecardResult;
+
+  return data;
+}
+
+export function buildURL(repository: string, plateform?: string) {
+  plateform ? undefined : (plateform = CONSTANTS.defaultPlatform);
+  return new URL(`${CONSTANTS.baseUrl}/${plateform}/${repository}`);
+}
```

Comment:
For me this method is not useful. And the platform argument is currently not used and cannot be used by the end user.